bgpy/simulation_engine/policies/aspa/aspa.py

```diff
@@ -40,104 +41,144 @@ def _valid_ann(self, ann: "Ann", from_rel: Relationships) -> bool:  # type: igno
+        else:
+            min_down_ramp = uppercase_j
 
-    def _calculate_u_min(self, ann: "Ann") -> int:
-        """Calculates u_min from ASPA RFC"""
-
-        path = ann.as_path
-
-        N = len(path)
-
-        def convert_u_to_index(u: int) -> int:
-            """Converts u to index in array
-
-            For example, take AS path 1, 2, 666, 777
-            In the ASPA RFC, this is  N, 3, 2,   1 indexed
-            And in Python, this is    0, 1, 2,   3 indexed
-            So to go from u to python index, you do:
-            """
-            return N - u
-
-        # https://datatracker.ietf.org/doc/html/draft-ietf-sidrops-aspa-verification-16
-        u_min = len(path) + 1  # type: ignore
-        for u in range(2, N + 1):
-            u_minus_one_index = convert_u_to_index(u - 1)
-            u_index = convert_u_to_index(u)
-            # False from this func is the same as "Not Provider+" from the ASPA RFC
-            if not self._provider_check(path[u_minus_one_index], path[u_index]):
-                u_min = u
-                break
-        return u_min
-
-    def _calculate_v_max(self, ann: "Ann") -> int:
-        """Calculates v_max from ASPA RFC"""
-
-        N = len(ann.as_path)
-
-        def convert_v_to_index(v: int) -> int:
-            """Converts v to index in array
-
-            For example, take AS path 1, 2, 666, 777
-            In the ASPA RFC, this is  N, 3, 2,   1 indexed
-            And in Python, this is    0, 1, 2,   3 indexed
-            So to go from v to python index, you do:
-            """
-            return N - v
-
-        v_max = 0
-        for v in range(N - 1, 1 - 1, -1):
-            v_index = convert_v_to_index(v)
-            v_plus_one_index = convert_v_to_index(v + 1)
-            # The provider check returning false is the same as the hop func
-            # described in the ASPA RFC section 5 returning "Not Provider+"
-            if not self._provider_check(
-                ann.as_path[v_plus_one_index], ann.as_path[v_index]
-            ):
-                v_max = v
-                break
-        return v_max
+        return min_down_ramp
 
     def _provider_check(self, asn1: int, asn2: int) -> bool:
-        """Returns False if asn2 is not in asn1's provider_asns, AND asn1 adopts ASPAi
+        """Returns True if asn2 is the provider of asn1. If asn1 does not adopt
+        ASPA, the function returns False. This function equals Provider+ from the RFC
 
-        This also essentially can take the place of the "hop check" listed in
-        ASPA RFC section 5
+        This function is used to establish the length of the min_up_ramp
+        and min_down_ramp
         """
-
         cur_as_obj = self.as_.as_graph.as_dict[asn1]
         if isinstance(cur_as_obj.policy, ASPA):
+            next_as_obj = self.as_.as_graph.as_dict[asn2]
+            return next_as_obj.asn in cur_as_obj.provider_asns
+        else:
+            return False
+
```

Comment:
provider_check and provider_or_peer check seem to be to be identical? I'm also a bit confused where the peers come into play?